src/runtime/pay-flow.js

```diff
@@ -166,26 +166,33 @@ export function validatePayResponse(port) {
  * @return {!SubscribeResponse}
  */
 export function parseSubscriptionResponse(data) {
+  let swgData = null;
   let raw = null;
   if (data) {
     if (typeof data == 'string') {
       raw = /** @type {string} */ (data);
     } else {
       // Assume it's a json object in the format:
-      // `{integratorClientCallbackData: "..."}`.
+      // `{integratorClientCallbackData: "..."}` or `{swgCallbackData: "..."}`.
       const json = /** @type {!Object} */ (data);
-      raw = json['integratorClientCallbackData'];
+      if ('swgCallbackData' in json) {
+        swgData = /** @type {!Object} */ (json['swgCallbackData']);
+      } else if ('integratorClientCallbackData' in json) {
+        raw = json['integratorClientCallbackData'];
+      }
     }
   }
-
-  let swgData = null;
-  if (raw) {
-    const parsed = parseJson(atob(raw));
-    swgData = parsed['swgCallbackData'];
+  if (raw && !swgData) {
+    raw = atob(raw);
+    if (raw) {
+      const parsed = parseJson(raw);
```

Comment:
Why not Json.Parse?

Reply:
That's pretty much that - with some nuances around types.

Reply:
Also. That call is inlined into `JSON.parse()` in the binary.